minio/minio
# Implement purge-on-delete feature for object deletions

If a special header - `x-minio-purge-on-delete` is passed in the DeleteObject API, the object
will be purged immediately and free up the space

## Description

this is a feature to immediately purge the object while deleting and freeing up the space from `.minio.sys/tmp/trash`

## Motivation and Context

to purge the deleted objects and reclaim the space

## How to test this PR?

- pass a special header `x-minio-purge-on-delete: true` in the delete object API 
- check if the corresponding entries are removed in `.minio.sys/tmp/trash`
- this minio-go PR https://github.com/minio/minio-go/pull/1644 can be used for testing

the benchmark tests by removing 100 entries reported the following

parallel deletes :-

- 0.0006639 ns/op
- memory profile reported around 1024.05 KB (18.05%)

sequential deletes :- 

- 12933971394 ns/op
- 22532.32 KB (79.70%)

## Types of changes

## Checklist:
